core/vm/privacy/bulletproof.go

```diff
@@ -935,9 +947,16 @@ func (mrp *MultiRangeProof) Deserialize(proof []byte) error {
 	offset += 32
 
 	numChallenges := int(math.Log2(float64(len(mrp.Comms)*bitsPerValue))) + 1
+	if len(proof) <= offset+numChallenges {
```

Comment:
here I think should be numChallenges*32 at least

looking inside `mrp.IPP.Deserialize(proof[offset:], numChallenges)` function
the added bytes length is same as line below `len(mrp.IPP.L)*33 + len(mrp.IPP.R)*33 + len(mrp.IPP.Challenges)*32 + 2*32`

there are also length checks in that function but it is incomplete

Reply:
i moved this implementation into the Deserialize method.